core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1637,10 +1644,10 @@ private Map<String, Argument> analyzeArguments(Node node, List<ArgumentSpecifica
                 }
             }
 
-            return passedArguments.buildOrThrow();
+            return new TableFunctionArgumentsAnalysis(passedArguments.buildOrThrow(), tableArgumentAnalyses.build());
         }
 
-        private Argument analyzeArgument(ArgumentSpecification argumentSpecification, TableFunctionArgument argument)
+        private Argument analyzeArgument(ArgumentSpecification argumentSpecification, TableFunctionArgument argument, Optional<Scope> scope, ImmutableList.Builder<TableArgumentAnalysis> tableArgumentAnalyses)
```

Comment:
I added a helper class to return both the `Argument`, and the optional `TableArgumentAnalysis`. `TableArgumentAnalysis` couldn't be extended to carry the `Argument`, because it only applies to table arguments, while `Argument` is created for each argument regardless of its type.